docs/subnets/introduction-to-vm.md

```diff
@@ -0,0 +1,335 @@
+
+### APIs for a VM
+
+We interact with a blockchain and underlying VM through **Static** and **Non-Static Handlers**. Handlers are used for making API calls to VM methods. Specifically, these handlers are implemented as **Services**.
+
+### Handlers
+
+Handlers serve the response for the incoming HTTP requests. Handlers can also be wrapped with **gRPC** for efficiently making calls from other services such as `AvalancheGo`. Handlers help in creating APIs. VM implements 2 kinds of handlers:
+
+- **Non-Static Handlers** - They help in interacting with blockchains instantiated by the VM. The API's endpoint will be different for different chains. `/ext/bc/[chainID]`
+- **Static Handlers** - They help in directly accessing VM. These are optional for reasons such as parsing genesis bytes required to instantiate new blockchains. `/ext/vm/[vmID]`
+
+For any developers familiar with object-oriented programming, this is very similar to static and non-static methods on a class.
+
+### VM Factory
+
+Each VM has a **factory** that is capable of creating new VM instances. A VM can be initialized as a blockchain along with handlers for accessing it, only when we have the VM's instance. The factory's `New` method shown below, returns the VM's instance to its caller in `AvalancheGo`. It's generally in the [`factory.go`](https://github.com/ava-labs/blobvm/blob/master/factory.go) file of the VM.
+
+```go
+// Returning a new VM instance from VM's factory
+func (f *Factory) New(*snow.Context) (interface{}, error) { return &vm.VM{}, nil }
+```
+
+### Initializing a VM
+
+Blockchains are functional, only when the instantiated VMs are initialized and the node is bootstrapped. Initializing a VM involves setting up the database, block builder, mempool, genesis state, handlers, etc. This will expose the VM's API handlers, start accepting transactions, gossiping them across the network, building blocks, etc. More details on it can be found in the [third part](/subnets/create-a-vm-blobvm#initialize) of this series.
+
+```go
+if err := vm.Initialize(
+    ctx.Context,
+    vmDBManager,
+    genesisData,
+    chainConfig.Upgrade,
+    chainConfig.Config,
+    msgChan,
+    fxs,
+    sender,
+);
+```
+
+You can refer to the [implementation](https://github.com/ava-labs/blobvm/blob/master/vm/vm.go#L92) of `vm.initialize` in the BlobVM repository.
+
+### Interaction
+
+We can use the API handlers to issue transactions, query chain state, and other functionalities provided by the VM.
+
+## Interfaces
+
+The main task of a VM is to issue blocks to the consensus engine whenever requested. It is the VM's responsibility to handle user transactions, and include them into new blocks, mempool handling, and database handling.
```

Comment:
I would not define this as the main task of the VM. The main task would be to verify blocks and handle accept/reject by performing the necessary state transitions.